### ISSUE TITLE
Deleted sensors visible in Home Assistant

### ISSUE THREAD

**Issue by Oliviakrkk** [dresden-elektronik/deconz-rest-plugin#1216]
Hi

I deleted sensors in Phoscon UI and checked via api (POST /api/<apikey>/sensors) that removed sensors are not visible anymore.

But sensors (cube and vibration) are still shown in home assistant:
![hass_deconz](https://user-images.githubusercontent.com/43586519/52202942-34df2c00-2870-11e9-8f1f-fa99be1c755c.PNG)

How can I remove those sensors?

**Comment by Kane610** [comment]
Have you tried restarting Home assistant?

**Comment by Oliviakrkk** [comment]
Yes.  Multiple times. And manually run services deconz.device_refresh. Neither helped. 

I even removed deconz integration and readded it - ghosts are still there.

**Comment by Kane610** [comment]
It should be. There are some comments in other issues about removing items from the database

**Comment by Oliviakrkk** [comment]
Looks like this solved the issue. Question is why API did not show those entries...

**Comment by Kane610** [comment]
It is marked as deleted, but for some reason the Rest API doesn't honor that

**Comment by manup** [comment]
The REST-API doesn't return sensors which are marked as deleted. Sensors which are marked as deleted in the database are further not loaded on startup.

So in theory these entries shouldn't make any trouble.

**Comment by forast3r** [comment]
@Oliviakrkk You can also delete the old (deleted) sensors in HA's core.device_registry

Be careful!

**Comment by Oliviakrkk** [comment]
Good point. If I encounter another ghost entries I will try that.

Thanks!

**Comment by snjoetw** [comment]
@forast3r I had to do this previously when I remove a device from deconz. 

But I had to also modify deconz DB to remove stale entries as well. I think this was fixed in recent release?

**Comment by Oliviakrkk** [comment]
I guess not, because I am on most recent one and I had issues.

**Comment by lbschenkel** [comment]
@Kane610: I can confirm that I have deleted devices showing in HASS (lights, sensors and switches) after I deleted them from deCONZ. I checked that they're not visible in the deCONZ REST API (confirmed with `curl`), they're not present in the deCONZ DB (rows for sensors were forcefully removed from the DB), I have restarted HASS multiple times, and yet these deleted devices show in the Config/Integrations page and remain in `core.device_registry`.

Should I file a bug in HASS?

P.S.: I tried to find the code that removes devices which are no longer exposed in the REST API from the HASS device registry to better understand the logic but I was not able to find it in https://github.com/home-assistant/home-assistant/tree/master/homeassistant/components/deconz. Did I miss it somewhere?
